main.rb

```diff
@@ -0,0 +1,175 @@
+# rubocop:disable Style/CyclomaticComplexity
+require './book'
+require './person'
+require './teacher'
+require './student'
+require './rental'
+
+class Library
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      print 'No books in library'
+      return
+    end
+    @books.each do |book|
+      print "Title: #{book.title.capitalize}, Author: #{book.author.capitalize}\n"
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp.capitalize
+    print 'Author: '
+    author = gets.chomp.capitalize
+    book = Book.new(title: title, author: author)
+    @books << book
+    puts "Book created successfully\n"
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'Your Library is empty'
+      return
+    end
+    @people.each do |person|
+      print "[#{person.class.name}] Name: #{person.name.capitalize}, ID: #{person.id}, Age: #{person.age}\n"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+
+    option = gets.chomp
+
+    case option
+    when '1'
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Classroom:'
+      classroom = gets.chomp
+      print 'Has parent permission? [Y/N]: '
+      permission_resp = gets.chomp
+      parent_permission = permission_resp.downcase == 'y'
+
```

Comment:
- [ ] Here parent permission is being asked both, student and teacher, but teacher should have parent permission set to true by default. Kindly render parent persission only for student, here is an example: 

```ruby
    person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
      end

    @people << person
    puts "Person created successfully"
  end
```